core-bundle/src/Doctrine/Backup/BackupManager.php

```diff
@@ -0,0 +1,265 @@
+{
+    public const DUMP_HEADER = '-- Generated by the Contao Open Source CMS Backup Manager (version: v1).';
+
+    private Connection $connection;
+    private DumperInterface $dumper;
+    private string $backupDir = '';
+    private array $tablesToIgnore = [];
+    private int $keepMax = 0;
+
+    public function __construct(Connection $connection, DumperInterface $dumper, string $backupDir, array $tablesToIgnore, int $keepMax)
+    {
+        $this->connection = $connection;
+        $this->dumper = $dumper;
+        $this->backupDir = $backupDir;
+        $this->tablesToIgnore = $tablesToIgnore;
+        $this->keepMax = $keepMax;
+    }
+
+    public function createCreateConfig(): CreateConfig
+    {
+        $config = new CreateConfig(Backup::createNewAtPath($this->backupDir));
+
+        return $config->withTablesToIgnore($this->tablesToIgnore);
+    }
+
+    /**
+     * @throws BackupManagerException
+     */
+    public function createRestoreConfig(): RestoreConfig
+    {
+        $latestBackup = $this->getLatestBackup();
+
+        if (null === $latestBackup) {
+            throw new BackupManagerException('No backups found.');
+        }
+
+        $config = new RestoreConfig($latestBackup);
+
+        return $config->withTablesToIgnore($this->tablesToIgnore);
+    }
+
+    public function getLatestBackup(): ?Backup
+    {
+        return $this->listBackups()[0] ?? null;
+    }
+
+    /**
+     * Lists all backups (latest one first).
+     *
+     * @return array<Backup>
+     */
+    public function listBackups(): array
+    {
+        (new Filesystem())->mkdir($this->backupDir); // Ensures, the directory exists
+
+        $backups = [];
+        $files = Finder::create()
+            ->files()
+            ->in($this->backupDir)
+            ->depth('== 0')
+            ->name(Backup::VALID_BACKUP_NAME_REGEX)
+        ;
+
+        foreach ($files as $file) {
+            $backups[] = new Backup($file->getPathname());
+        }
+
+        uasort($backups, static fn (Backup $a, Backup $b) => $b->getCreatedAt() <=> $a->getCreatedAt());
+
+        return array_values($backups);
+    }
+
+    /**
+     * @throws BackupManagerException
+     */
+    public function create(CreateConfig $config): void
+    {
+        $this->executeTransactional(
+            function () use ($config): void {
+                $this->doCreate($config);
+            }
+        );
+    }
+
+    /**
+     * @throws BackupManagerException
+     */
+    public function restore(RestoreConfig $config): void
+    {
+        $this->executeTransactional(
+            function () use ($config): void {
+                $this->doRestore($config);
+            }
+        );
+    }
+
+    /**
+     * @throws BackupManagerException
+     */
+    private function executeTransactional(\Closure $func): void
+    {
+        $isAutoCommit = $this->connection->isAutoCommit();
+
+        if ($isAutoCommit) {
+            $this->connection->setAutoCommit(false);
+        }
```

Comment:
This was the only way I could get a proper rollback to work while trying to actually backup a DB.

Reply:
The backup process is read only, so commiting a transaction (either explicitly or by setting auto commit) would just release the locks, no? You cannot rollback imports anyways because `MySQL` does not support transactions for DDL queries.

Reply:
I lack the knowledge here so I'd love to have you checkout the branch and do some tests yourself. It worked for me 🙈

Reply:
Let's leave this as is. I added it to my list + can still make another PR if we want to change something later on.